rust/headless-client/src/uptime_lib/mod.rs

```diff
@@ -0,0 +1,85 @@
+// Manually vendored from https://github.com/itchyny/uptime-rs/blob/e6e31f4aa69b057d0610c20f8f2b5f8f0724decb/src/lib.rs and https://github.com/itchyny/uptime-rs/blob/e6e31f4aa69b057d0610c20f8f2b5f8f0724decb/tests/lib.rs
+
+use std::time::Duration;
+
+#[derive(Debug, thiserror::Error)]
+#[cfg_attr(target_os = "windows", allow(unused))]
+pub enum Error {
+    #[cfg(target_os = "linux")]
+    #[error("sysinfo failed")]
+    Sysinfo,
+
+    #[cfg(any(
+        target_os = "macos",
+        target_os = "freebsd",
+        target_os = "openbsd",
+        target_os = "netbsd"
+    ))]
+    #[error("sysctl failed")]
+    Sysctl,
+
+    #[cfg(any(
+        target_os = "macos",
+        target_os = "freebsd",
+        target_os = "openbsd",
+        target_os = "netbsd"
+    ))]
+    #[error(transparent)]
+    SystemTime(#[from] std::time::SystemTimeError),
+}
+
+#[cfg(target_os = "linux")]
+pub fn get() -> Result<Duration, Error> {
```

Comment:
Once the other one is a stub, we can make this an `Option` and delete the `Error`.